### PR TITLE
stop relying on legacy APIs in some integrations

### PR DESCRIPTION
**Pull request by sryza**
### Summary & Motivation

This is by no means comprehensive - just a set of changes I made to get a feel for it.

I think it's fine if some error messages say "op" instead of solid for legacy APIs for the next couple releases.

### CODE REVIEW

File: python_modules/libraries/dagster-airflow/dagster_airflow/dagster_pipeline_factory.py [hunk 1]
```diff
@@ -399,7 +398,7 @@ def make_dagster_solid_from_airflow_task(task, use_airflow_template_context, uni
     check.bool_param(use_airflow_template_context, "use_airflow_template_context")
     unique_id = check.opt_int_param(unique_id, "unique_id")
 
-    @solid(
+    @op(
         name=normalized_name(task.task_id, unique_id),
         input_defs=[InputDefinition("airflow_task_ready", Nothing)],
         output_defs=[OutputDefinition(Nothing, "airflow_task_complete")],
```
Comment (dpeng817) [inline python_modules/libraries/dagster-airflow/dagster_airflow/dagster_pipeline_factory.py:404]: needs to get rid of InputDefinition/OutputDefinition as well

File: python_modules/libraries/dagster-dbt/dagster_dbt/cli/solids.py [hunk 2]
```diff
@@ -34,7 +34,7 @@ def passthrough_flags_only(solid_config, additional_flags):
     }
 
 
-@solid(
+@op(
     description="A solid to invoke dbt run via CLI.",
     input_defs=[InputDefinition(name="start_after", dagster_type=Nothing)],
     output_defs=[OutputDefinition(name="dbt_cli_output", dagster_type=DbtCliOutput)],
```
Comment (dpeng817) [inline python_modules/libraries/dagster-dbt/dagster_dbt/cli/solids.py:40]: InputDefinition/OutputDefinition

File: python_modules/libraries/dagster-dbt/dagster_dbt/cli/solids.py [hunk 3]
```diff
@@ -125,7 +125,7 @@ def dbt_cli_run(context):
     yield Output(cli_output, "dbt_cli_output")
 
 
-@solid(
+@op(
     description="A solid to invoke dbt test via CLI.",
     input_defs=[InputDefinition(name="start_after", dagster_type=Nothing)],
     output_defs=[OutputDefinition(name="dbt_cli_output", dagster_type=DbtCliOutput)],
```
Comment (dpeng817) [inline python_modules/libraries/dagster-dbt/dagster_dbt/cli/solids.py:131]: same as [1]

File: python_modules/libraries/dagster-dbt/dagster_dbt/cli/solids.py [hunk 4]
```diff
@@ -202,7 +202,7 @@ def dbt_cli_test(context):
     yield Output(cli_output, "dbt_cli_output")
 
 
-@solid(
+@op(
     description="A solid to invoke dbt snapshot via CLI.",
     input_defs=[InputDefinition(name="start_after", dagster_type=Nothing)],
     output_defs=[OutputDefinition(name="dbt_cli_output", dagster_type=DbtCliOutput)],
```
Comment (dpeng817) [inline python_modules/libraries/dagster-dbt/dagster_dbt/cli/solids.py:208]: etc

**Review by dpeng817 — CHANGES_REQUESTED** [review]
Good start, some of them still need InputDefinition/OutputDefinition migrated. Thanks for getting ahead on this.

**Comment by sryza** [comment]
@dpeng817 this isn't meant to be comprehensive. For the input / output defs that aren't done yet, I'm thinking we can do those later.

**Review by dpeng817 — APPROVED** [review]

**Pull request merged**